### ISSUE TITLE
invalid yaml produces cryptic error

### ISSUE THREAD

**Issue by jpcarey** [elastic/logstash#6509]
Should improve the error thrown when an invalid yaml file is read from logstash.yml.

Current error:
`ERROR: Failed to load settings file from "path.settings". Aborting... path.setting=/etc/logstash/, exception=Psych::SyntaxError, message=>(<unknown>): expected <block end>, but found BlockMappingStart while parsing a block mapping at line 155 column 2`

- Version: 5.1.1
- Operating System: Cetos 7
- Steps to Reproduce:
  - Install RPM
  - In /etc/logstash/logstash.yml, uncomment line 155, leaving the space at the beginning of the line.
  - Start logstash, and you will get the error pasted above.

**Comment by jsvd** [comment]
We're limited to what the ruby's yaml parser gives us:

```
jruby-1.7.25 :017 > YAML.parse("teste:\n    teste.tet:\n  hey: - 1\n")
Psych::SyntaxError: (<unknown>): expected <block end>, but found BlockMappingStart while parsing a block mapping at line 3 column 3
```

The only option is to create a user friendly component that parses these messages, but it seems like a big time investment ..

**Comment by jpcarey** [comment]
I think if the error was more descriptive - that `YAML.parse` failed, and which file - that would be more than sufficient (along with the raw exception, which does contain the line number and character).

Maybe something like:
`ERROR: Failed to parse YAML file: {{path.to.file}}. Aborting... path.setting=/etc/logstash/, exception=Psych::SyntaxError, message=>(<unknown>): expected <block end>, but found BlockMappingStart while parsing a block mapping at line 155 column 2`

**Comment by jsvd** [comment]
@jpcarey makes sense, thank you for the suggestion

**Comment by suyograo** [comment]
Fixed in #6546
